From d37b2f8bc6b697c11a2c7e326649bf2cedddf1f6 Mon Sep 17 00:00:00 2001
From: Marvin Ramin <marvinramin@google.com>
Date: Wed, 16 Oct 2024 12:26:02 +0200
Subject: [PATCH] RESTRICT AUTOMERGE Use getLaunchedFromPackage instead of
 getCallingPackage

Ensure that MediaProjectionPermissionActivity is doing proper
attribution and validation of permissions before showing.

Bug: 372670004
Bug: 373581993
Test: presubmit
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:39ff0377a592fa9bacd308e982a397106665a046)
Merged-In: I48adb14fbc8d5f2f2e21ef8a30a713d33f03ab02
Change-Id: I48adb14fbc8d5f2f2e21ef8a30a713d33f03ab02
---
 .../systemui/media/MediaProjectionPermissionActivity.java    | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/media/MediaProjectionPermissionActivity.java b/packages/SystemUI/src/com/android/systemui/media/MediaProjectionPermissionActivity.java
index f6a2f3704283..eb7698c42329 100644
--- a/packages/SystemUI/src/com/android/systemui/media/MediaProjectionPermissionActivity.java
+++ b/packages/SystemUI/src/com/android/systemui/media/MediaProjectionPermissionActivity.java
@@ -98,12 +98,11 @@ public class MediaProjectionPermissionActivity extends Activity
         final Intent launchingIntent = getIntent();
         mReviewGrantedConsentRequired = launchingIntent.getBooleanExtra(
                 EXTRA_USER_REVIEW_GRANTED_CONSENT, false);
-
-        mPackageName = getCallingPackage();
+        mPackageName = getLaunchedFromPackage();
 
         // This activity is launched directly by an app, or system server. System server provides
         // the package name through the intent if so.
-        if (mPackageName == null) {
+        if (getCallingPackage() == null)  {
             if (launchingIntent.hasExtra(EXTRA_PACKAGE_REUSING_GRANTED_CONSENT)) {
                 mPackageName = launchingIntent.getStringExtra(
                         EXTRA_PACKAGE_REUSING_GRANTED_CONSENT);
-- 
2.47.1.613.gc27f4b7a9f-goog

